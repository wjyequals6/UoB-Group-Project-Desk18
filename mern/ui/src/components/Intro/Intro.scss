
@import "../../common.scss";

@keyframes shake {
    0% { transform: translate(1px, 1px) rotate(0deg); }
    10% { transform: translate(-1px, -2px) rotate(-1deg); }
    20% { transform: translate(-3px, 0px) rotate(1deg); }
    30% { transform: translate(3px, 2px) rotate(0deg); }
    40% { transform: translate(1px, -1px) rotate(1deg); }
    50% { transform: translate(-1px, 2px) rotate(-1deg); }
    60% { transform: translate(-3px, 1px) rotate(0deg); }
    70% { transform: translate(3px, 1px) rotate(-1deg); }
    80% { transform: translate(-1px, -1px) rotate(1deg); }
    90% { transform: translate(1px, 2px) rotate(0deg); }
    100% { transform: translate(1px, -2px) rotate(-1deg); }
}

@keyframes opening {
    0% { opacity: 0;}
    100% { opacity: 1; }
}

@keyframes enlarge {
    0%{transform: scale(1,1);}
    100%{transform: scale(1.05,1.05);}
}

@keyframes bounce {
	0%, 20%, 50%, 80%, 100% {transform: translateY(0);}
	40% {transform: translateY(-30px);}
	60% {transform: translateY(-15px);}
}

.Intro{
    h1{
        font-weight: 400;
        font-size: 20px;
        color: #fff;
        color: #fff;
    }
    position: fixed;
    top: 0;
    left: 0;
    box-sizing: border-box;
    width: 100%;
    height: 100%;
    background-color: $primary;
    opacity: 1;
    z-index: 200;
    transition: 1s;
    @include flexCenter;
    align-items: center;
    &.showIntro{
        opacity: 1;
        visibility: visible;
    }
    &.closeIntro{
        opacity: 0;
        visibility: hidden;
    }

    .btnText{
        user-select: none;
        cursor: pointer;
        transition: 1s;
        margin: 0 auto;
        @include flexCenter;
        // text-decoration: underline;
        font-weight: 400;
        font-size: 20px;
        position: relative;
        color: #fff;
        &:after{
            content:"";
            width: 0;
            height: 2px;
            background-color: #fff;
            position: absolute;
            bottom: 0;
            left: 50%;
            transform: translateX(-50%);
            transition: all 0.3s;
          }
        &:hover:after{
            width: 100%;
          }
    }   

    .IntroLogo{
        transition: 0.5s;
        position: fixed;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 100%;
        height: 100%;
        animation: opening 1.5s 0s linear;
        @include flexCenter;
        align-items: center;
        flex-direction: column;
        .logo{
            user-select: none;
            animation: bounce 2s ease;
            cursor: pointer;
            transition: 1s;
            box-sizing: border-box;
            width: 16vw;
            height: 16vw;
            margin: 0 auto;
            @include flexCenter;
            align-items: center;
            &:hover{
                animation: bounce 2s ease infinite;
            }
        }
        .logoTxt{
            user-select: none;
            cursor: pointer;
            transition: 1s;
            box-sizing: border-box;
            width: 42vw;
            height: 5vw;
            @include flexCenter;
            align-items: center;
        }
    }

    .Slide{
        transition: 0.5s;
        transition: 0.5s;
        position: fixed;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 100%;
        height: 100%;
        .ant-carousel .slick-slide {
            @include flexCenter;
            align-items: center;
            text-align: center;
            height: 90vh;
            background: #244949;
            overflow: hidden;
            @include flexCenter;
            align-items: center;
        }
        .slideTextBox{
            padding-top: 20px;
            margin-left: 20px;
            margin-right: 20px;
        }
        .introSlide{
            box-sizing: border-box;
            margin: 0 auto;
            width: 40vw;
            height: 30vw;
            max-width: 600px;
            max-height: 360px;
            &.logoSlide{
                width: 38vw;
                height: 22vw;
                max-width: 520px;
                max-height: 290px;
                padding-bottom: 10px;
            }
            &.introSlide0{
                width: 30vw;
                max-width: 360px;
            }
            &.introSlide2{
                height: 24vw;
                max-height: 320px;
            }
        }
        .iconLeft{
            position: absolute;
            left: 10vw;
            top: 50%;
            transform: translateY(-50%);
            font-size: 30px;
            color: #fff;
            z-index: 300;
        }
        .iconRight{
            position: absolute;
            right: 10vw;
            top: 50%;
            transform: translateY(-50%);
            font-size: 30px;
            color: #fff;
            z-index: 300;
        }
        .btnSkip{
            user-select: none;
            position: absolute;
            left: 50%;
            bottom: 12vh;
            transform: translate(-50%,-50%);
            cursor: pointer;
            transition: 1s;
            font-weight: 400;
            font-size: 20px;
            color: #fff;
            &:after{
                content:"";
                width: 0;
                height: 2px;
                background-color: #fff;
                position: absolute;
                bottom: 0;
                left: 50%;
                transform: translateX(-50%);
                transition: all 0.3s;
            }
            &:hover:after{
                width: 100%;
            }
        }
    }

    

    
    
}