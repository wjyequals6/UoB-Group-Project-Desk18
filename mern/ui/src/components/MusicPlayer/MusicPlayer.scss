@import "../../common.scss";
.MusicPlayer{
    opacity: 1;
    position: fixed;
    z-index: 300;
    top: 88%;
    left: 90%;
}

  .player__button {
    width: fit-content;
    margin-bottom: 15px;
    background-color: transparent;
    border: none;

    &:focus {
      outline: none;
    }
    &:hover {
      cursor: pointer;
      svg {
        color: green;
      }
    }

    svg {
      font-size: 4em;
      color:rgba(0, 0, 0, 0.5);
    }
  }

  