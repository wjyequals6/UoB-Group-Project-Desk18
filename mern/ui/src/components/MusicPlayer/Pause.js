import React from "react";
import { PauseCircleOutlined } from '@ant-design/icons';


export default function Play(props) {
  const { handleClick } = props;
  
  return (
    <button title="btnPause" className="player__button" onClick={() => handleClick()}>
      <PauseCircleOutlined />
    </button>
  );
}
