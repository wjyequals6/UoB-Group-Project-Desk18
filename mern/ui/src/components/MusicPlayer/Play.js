import React from "react";
import { PlayCircleOutlined } from '@ant-design/icons';
export default function Play(props) {
  const { handleClick } = props;

  return (
    <button title="btnPlay" className="player__button" onClick={() => handleClick()}>
      <PlayCircleOutlined />
    </button>
  );
}
