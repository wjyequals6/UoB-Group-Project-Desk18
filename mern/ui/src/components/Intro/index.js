import React, { useState, useEffect } from "react";
import "./Intro.scss";
import logo_removebg from "../../img/logo_removebg.png";
import logoTxt_removebg from "../../img/logoTxt_removebg.png";
import logoSlide from "../../img/logoSlide.png";
import introSlide0 from "../../img/introSlide0.png";
import introSlide1 from "../../img/introSlide1.png";
import introSlide2 from "../../img/introSlide2.gif";
import introSlide3 from "../../img/introSlide3.gif";


import { Carousel } from 'antd';
import { LeftOutlined, RightOutlined } from '@ant-design/icons';
import 'antd/dist/antd.css';

const Intro = (props) => {
    const [showIntro, setShowIntro] = useState(true);
    const [showLogo, setShowLogo] = useState(true);
    const [showSlide, setShowSlide] = useState(false);

    

    const carouselRef = React.createRef();

    const handleNext = () => carouselRef.current.next();

    const handlePrev = () => carouselRef.current.prev();

    const {  } = props;
    return (
       <div className="Intro" data-testid="Intro" style={showIntro ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}  >
           <div className="IntroLogo" style={showLogo ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}>
                <img src={logo_removebg} data-testid="logo" className="logo" onClick={() => {setShowLogo(false);setShowSlide(true); }} />
                <img src={logoTxt_removebg} data-testid="logoTxt" className="logoTxt" onClick={() => {setShowLogo(false);setShowSlide(true); }} />
                <div><div className="btnText startText" data-testid="btnStart" onClick={() => {setShowIntro(false); setShowLogo(false); setShowSlide(false);}}>START</div></div>
                <div><div className="btnText tutorialText" data-testid="btnTutorial" onClick={() => {setShowLogo(false);setShowSlide(true); }}>TUTORIAL</div></div>
           </div>
           <div className="Slide" style={showSlide ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}>
                <LeftOutlined className="iconLeft" onClick={handlePrev}  />     
                <Carousel ref={carouselRef}>
                    <div>
                        <img className="introSlide logoSlide" src={logoSlide}></img>
                        <div className="slideTextBox"><h1>COVIDSURVIVAL IS A PLATFORM FOR EVERYONE TO LEARNING THE INFROMATION OF COVID-19 AND SHARE YOUR THOUGHTS</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide0" src={introSlide0}></img>
                        <div className="slideTextBox"><h1>THE GLOBAL STATICS ARE SHOWN ON THE TOP LEFT OF THE PAGE</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide1" src={introSlide1}></img>
                        <div className="slideTextBox"><h1>ONE CLICK ON MAP FOR REAL-TIME OFFICIAL STATISTICS OF COVID-19</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide2" src={introSlide2}></img>
                        <div className="slideTextBox"><h1>FURTHER DETAILS OF TRAVEL POLICY AND NEWS ARE ON THE BOTTOM-LEFT OF THE PAGE</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide3" src={introSlide3}></img>
                        <div className="slideTextBox"><h1>AFTER SIGN-IN, YOU CAN ACCESS THE FORUM TO SHARE THE TIPS TO SURVIVE THE COVID</h1></div>
                    </div>
               </Carousel>
               <RightOutlined className="iconRight" onClick={handleNext}/>
               <div><div className="btnSkip" data-testid="btnSkip" onClick={() => {setShowIntro(false); setShowLogo(false); setShowSlide(false);}}>SKIP AND START</div></div>
           </div>
       </div>
    );
};

export default Intro;