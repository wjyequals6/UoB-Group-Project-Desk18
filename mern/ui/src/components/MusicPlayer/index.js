import React from "react";
import { useState, useEffect } from "react";
import "./MusicPlayer.scss";

import Play from "./Play";
import Pause from "./Pause";
import theme from "../../music/themeMusic.mp3";
import buttonSound from "../../music/button.mp3";

function MusicPlayer() {
  const [audioButton] = useState(new Audio(buttonSound));
  const [playing, setPlaying] = useState(true);
  const [volume] = useState(30);
  const finalVolume = (volume / 100) ** 2

  useEffect(() => {
    const audio = document.getElementById("audio");
    if(audio != null){
      audio.volume = finalVolume;
      audio.loop = true;
      playing ? audio.play() : audio.pause();
    }
    
  });

  useEffect(() => {
    
         document.body.addEventListener('mousedown', soundEffect );

      return function cleanup() {
          window.removeEventListener('mousedown', soundEffect );
      } 
    
  },[]);

  let soundEffect = () => {
    audioButton.volume = 0.7;
    audioButton.play();
  }
  

  
  return (
    <div className="player MusicPlayer" title="MusicPlayer">
      <audio id="audio">
        <source src={theme} />
        Your browser does not support the <code>audio</code> element.
      </audio>
      
      <div className="controls">
        {playing ? 
          <Pause  handleClick={() => setPlaying(false)} /> :
          <Play data-testid="btnPlay" handleClick={() => setPlaying(true)} />
        }
      </div>
    </div>
  );
}

export default MusicPlayer;