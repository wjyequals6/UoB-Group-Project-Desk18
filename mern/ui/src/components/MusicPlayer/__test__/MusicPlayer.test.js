import MusicPlayer from '../index';
import React from 'react';
import ReactDOM from 'react-dom';
import Intro from './../index';
import { render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import userEvent from '@testing-library/user-event';


window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
window.HTMLMediaElement.prototype.addTextTrack = () => { /* do nothing */ };

afterEach(cleanup); 

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<MusicPlayer/>, div);
});

it("check music button playing state", ()=>{
    const {queryByTitle} = render(<MusicPlayer/>);
    expect(queryByTitle('btnPlay')).toBe(null);
    expect(queryByTitle('btnPause')).toBeTruthy();
    fireEvent.click(queryByTitle('btnPause'));
    expect(queryByTitle('btnPlay')).toBeTruthy();
    expect(queryByTitle('btnPause')).toBe(null);
    fireEvent.click(queryByTitle('btnPlay'));
    expect(queryByTitle('btnPlay')).toBe(null);
    expect(queryByTitle('btnPause')).toBeTruthy();
});


it("match snapshot", () => {
    const tree = renderer.create(<MusicPlayer/>).toJSON();
    expect(tree).toMatchSnapshot();
});