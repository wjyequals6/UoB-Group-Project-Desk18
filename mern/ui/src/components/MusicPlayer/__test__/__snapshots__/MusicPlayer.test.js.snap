// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="player MusicPlayer"
  title="MusicPlayer"
>
  <audio
    id="audio"
  >
    <source
      src="themeMusic.mp3"
    />
    Your browser does not support the 
    <code>
      audio
    </code>
     element.
  </audio>
  <div
    className="controls"
  >
    <button
      className="player__button"
      onClick={[Function]}
      title="btnPause"
    >
      <span
        aria-label="pause-circle"
        className="anticon anticon-pause-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="pause-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm-88-532h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8zm224 0h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
